akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -518,19 +603,127 @@ final class ClusterReceptionistSettings(
+  }
+
+  // END BINARY COMPATIBILITY
+
   private def copy(
     role: Option[String] = role,
     numberOfContacts: Int = numberOfContacts,
-    responseTunnelReceiveTimeout: FiniteDuration = responseTunnelReceiveTimeout): ClusterReceptionistSettings =
-    new ClusterReceptionistSettings(role, numberOfContacts, responseTunnelReceiveTimeout)
-
+    responseTunnelReceiveTimeout: FiniteDuration = responseTunnelReceiveTimeout,
+    heartbeatInterval: FiniteDuration = heartbeatInterval,
+    acceptableHeartbeatPause: FiniteDuration = acceptableHeartbeatPause,
+    failureDetectionInterval: FiniteDuration = failureDetectionInterval): ClusterReceptionistSettings =
+    new ClusterReceptionistSettings(
+      role,
+      numberOfContacts,
+      responseTunnelReceiveTimeout,
+      heartbeatInterval,
+      acceptableHeartbeatPause,
+      failureDetectionInterval)
 }
 
 /**
  * Marker trait for remote messages with special serializer.
  */
 sealed trait ClusterClientMessage extends Serializable
 
+/**
+ * Declares a super type for all events emitted by the `ClusterReceptionist`.
+ * in relation to cluster clients being interacted with.
+ * Events are published to the Akka event stream (system.eventStream). Given
+ * that these messages are transmitted across the event bus there is no need
+ * for serialization.
+ */
+sealed trait ClusterClientInteraction {
+  val clusterClient: ActorRef
+  val clusterReceptionist: ActorRef
+}
+
+/**
+ * Emitted to the Akka event stream when a cluster client has interacted with
+ * a receptionist.
+ */
+final case class ClusterClientUp(
+  override val clusterClient: ActorRef,
+  override val clusterReceptionist: ActorRef) extends ClusterClientInteraction
+
+/**
+ * Emitted to the Akka event stream when a cluster client was previously connected
+ * but then not seen for some time.
+ */
+final case class ClusterClientUnreachable(
+  override val clusterClient: ActorRef,
+  override val clusterReceptionist: ActorRef) extends ClusterClientInteraction
+
+abstract class GetClusterClients
+/**
+ * Get the cluster clients known to this receptionist. A ``ClusterClients`` message
+ * will be replied. A restriction of this message is that it can only be requested
+ * from within the same actor system as the receptionist (which is generally what
+ * you want when listening to it given that the event bus being used in tandem is
+ * only for the local actor system also).
+ */
+case object GetClusterClients extends GetClusterClients {
+  /**
+   * Java API: get the singleton instance
+   */
+  def getInstance = this
+}
+
+/**
+ * The reply to ``GetClusterClients``.
+ *
+ * @param clusterClients The presently known list of cluster clients.
+ */
+final case class ClusterClients(clusterClients: Set[ActorRef]) extends NoSerializationVerificationNeeded {
+  import scala.collection.JavaConverters._
+
+  /**
+   * Java API
+   */
+  val getClusterClients: java.util.Set[ActorRef] =
```

Comment:
Now a def, but still needs the NoSerializationVerificationNeeded.

> On 11 May 2016, at 7:17 PM, Patrik Nordwall <notifications@github.com> wrote:
> 
> In akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala <https://github.com/akka/akka/pull/20462#discussion_r62815466>:
> 
> > +   */
> > +  def getInstance = this
> > +}
> > +
> > +/**
> > + * The reply to ``GetClusterClients``.
> > + *
> > + * @param clusterClients The presently known list of cluster clients.
> > + */
> > +final case class ClusterClients(clusterClients: Set[ActorRef]) extends NoSerializationVerificationNeeded {
> > +  import scala.collection.JavaConverters._
> > +
> > +  /**
> > +   * Java API
> > +   */
> > +  val getClusterClients: java.util.Set[ActorRef] =
> please use def, and then you will not need NoSerializationVerificationNeeded
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/akka/akka/pull/20462/files/e2b0b1839ba16d6b07141a6955da1ab8e15cd412#r62815466>